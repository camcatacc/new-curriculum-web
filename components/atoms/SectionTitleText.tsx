import { Typography } from "@material-ui/core";
import React from "react";

export const SectionTitleText = (props: { children: string }) => {
    return (
        <Typography variant="h2" align="center">
            {props.children}
        </Typography>
    );
};
