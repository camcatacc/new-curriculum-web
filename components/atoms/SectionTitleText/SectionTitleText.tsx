// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";
import { getMotionFromOutsideVariants } from "utils/getMotionFromOutsideOptions";
import { motion } from "framer-motion";

// Definitions
import type { MotionProps } from "framer-motion";

export interface SectionTitleTextProps {
	children: string;
	animated?: boolean;
}

// Auxiliary functions
const getMotionOptions = (animated?: boolean, delay?: number): MotionProps => {
	if (!animated) return {};
	const { initial, animation } = getMotionFromOutsideVariants("fromLeft", delay);
	return {
		variants: { initial, animation },
		initial: "initial",
		whileInView: "animation",
		viewport: {
			once: true,
			amount: 1
		}
	};
};

// Element
const SectionTitleText = ({ children, animated }: SectionTitleTextProps) => {
	const motionOptions = getMotionOptions(animated);
	return (
		<motion.div {...motionOptions}>
			<Typography variant="h2" align="center" className="uppercase">
				{children}
			</Typography>
		</motion.div>
	);
};

export default SectionTitleText;
