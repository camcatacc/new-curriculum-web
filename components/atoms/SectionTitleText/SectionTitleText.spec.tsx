// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import SectionTitleText from "components/atoms/SectionTitleText/SectionTitleText";

// Definitions
import type { SectionTitleTextProps } from "components/atoms/SectionTitleText/SectionTitleText";

// Default Props
const defaultProps: SectionTitleTextProps = {
	children: "Bu"
};

// Tests
describe("SectionTitleText", () => {
	it("Renders as expected", () => {
		const { container } = render(<SectionTitleText {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
