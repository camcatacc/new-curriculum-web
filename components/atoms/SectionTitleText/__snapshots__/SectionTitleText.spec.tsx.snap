// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitleText Renders as expected 1`] = `
<div>
  <div>
    <h2
      class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
    >
      Bu
    </h2>
  </div>
</div>
`;
